contrib/checkbox-ce-oem/checkbox-provider-ce-oem/bin/rpmsg_load_firmware.py

```diff
@@ -0,0 +1,272 @@
+        )
+        self._rpmsg_state = Path(
+            "/sys/class/remoteproc/{}/state".format(remoteproc_dev)
+        )
+        self._search_patterns = {}
+        self.expected_events = []
+
+    def __enter__(self):
+        self._setup()
+        return self
+
+    def __exit__(self, type, value, traceback):
+        self._teardown()
+
+    def _setup(self):
+        self._previous_config = OrderedDict()
+        for key in self.properties:
+            self._previous_config.update({key: getattr(self, key)})
+
+    def _teardown(self):
+        self.rpmsg_state = "stop"
+        for key in self.properties:
+            if getattr(self, key) != self._previous_config[key]:
+                setattr(self, key, self._previous_config[key])
+
+    @property
+    def firmware_path(self) -> str:
+        return self._firmware_path.read_text()
+
+    @firmware_path.setter
+    def firmware_path(self, value: str) -> None:
+        self._firmware_path.write_text(value)
+
+    @property
+    def firmware_file(self) -> str:
+        return self._firmware_file.read_text()
+
+    @firmware_file.setter
+    def firmware_file(self, value: str) -> None:
+        self._firmware_file.write_text(value)
+
+    @property
+    def rpmsg_state(self) -> str:
+        return self._rpmsg_state.read_text()
+
+    @rpmsg_state.setter
+    def rpmsg_state(self, value: str) -> None:
+        self._rpmsg_state.write_text(value)
+
+    @property
+    def search_pattern(self) -> dict:
+        return self._search_patterns
+
+    @search_pattern.setter
+    def search_pattern(self, patterns: dict) -> None:
+        self._search_patterns.update(patterns)
+
+    def _init_logger(self):
+        self.log_reader = journal.Reader()
+        self.log_reader.this_boot()
+        self.log_reader.seek_tail()
+        self.log_reader.get_previous()
+
+        self._poller = select.poll()
+        self._poller.register(self.log_reader, self.log_reader.get_events())
+
+    def lookup_reload_logs(self, entry: dict):
+        keep_looking = True
+        for key, pattern in self._search_patterns.items():
+            if re.search(pattern, entry.get("MESSAGE")):
+                self.expected_events.append((key, entry.get("MESSAGE")))
+                if key == "ready":
+                    keep_looking = False
+                    break
+
+        return keep_looking
+
+    def _monitor_journal_logs(self, lookup_func):
+        start_time = time.time()
+        logging.info("# start time: %s", start_time)
+
+        while self._poller.poll(1000):
+            if self.log_reader.process() != journal.APPEND:
+                continue
+            for entry in self.log_reader:
+                logging.debug(entry["MESSAGE"])
+                if entry["MESSAGE"] == "":
+                    continue
+                if lookup_func(entry) is False:
+                    return self.expected_events
+
+            cur_time = time.time()
+            if (cur_time - start_time) > 60:
```

Comment:
May I know the reason the delta is 60?

Reply:
In typical scenarios, the reload firmware test is expected to conclude within 10 seconds. However, given the possibility of executing these tests on systems with lower performance, I have extended the timeout to 60 seconds.